File: python-gdb-activities-20/activity20/gdb/logging/memory_log_destination.py
# gdb/logging/memory_log_destination.py
from typing import List
from gdb.domain.transaction import Transaction
from gdb.logging.log_destination import LogDestination

class MemoryLogDestination(LogDestination):
    """In-memory storage destination implementor."""

    def __init__(self) -> None:
        self._transactions: List[Transaction] = []

    def write(self, transaction: Transaction) -> None:
        self._transactions.append(transaction)

    def read_all(self) -> List[Transaction]:
        return list(self._transactions)

    def get_destination_name(self) -> str:
        return "In-Memory"

    def clear(self) -> None:
        self._transactions.clear()

File: java-gdb-activities-20/activity20-solution/src/com/gdb/logging/MemoryLogDestination.java
package com.gdb.logging;

import com.gdb.command.TransactionCommand;
import java.util.*;

public class MemoryLogDestination implements LogDestination {
    private final List<TransactionCommand> commands = new ArrayList<>();

    @Override
    public void write(TransactionCommand cmd) {
        commands.add(cmd);
    }

    @Override
    public List<TransactionCommand> readAll() {
        return new ArrayList<>(commands);
    }

    @Override
    public void clear() {
        commands.clear();
    }

    @Override
    public String getDestinationName() {
        return "MEMORY";
    }
}
